frameworks/C++/ltio/http_benchmark.cc

```diff
@@ -0,0 +1,98 @@
+#include <vector>
+#include <csignal>
+
+#include "fmt/chrono.h"
+#include "gflags/gflags.h"
+#include "base/utils/string/str_utils.h"
+#include "net_io/server/http_server/http_server.h"
+
+
+using namespace lt::net;
+using namespace lt;
+using namespace base;
+
+DEFINE_int32(loops, 4, "how many loops use for handle message and io");
+DEFINE_string(http, "0.0.0.0:5006", "host:port used for http service listen on");
+
+class HttpBenchMarkServer {
+public:
+  ~HttpBenchMarkServer() {
+    LOG(INFO) << __func__ << " close app";
+    for (auto loop : loops) {
+      delete loop;
+    }
+    loops.clear();
+  }
+
+  void Run() {
+    main_loop.Start();
+
+    int loop_count = std::max(FLAGS_loops, int(std::thread::hardware_concurrency()));
+    LOG(INFO) << __func__ << " use loop count:" << loop_count;
+
+    for (int i = 0; i < loop_count; i++) {
+      auto loop = new(base::MessageLoop);
+      loop->Start();
+      loops.push_back(loop);
+    }
+
+    http_server.WithIOLoops(loops)
+      .WithAddress(base::StrUtil::Concat("http://", FLAGS_http))
+      .ServeAddress([](const RefHttpRequestCtx& context) {
+        VLOG(GLOG_VTRACE) << " got Http request from benchmark api";
+
+        const HttpRequest* req = context->Request();
+        //TODO: response freelist
+        auto response = HttpResponse::CreateWithCode(200);
+        response->SetKeepAlive(req->IsKeepAlive());
+
+        response->InsertHeader("Server", "ltio");
+        auto tm = fmt::gmtime(std::time(nullptr));
+        response->InsertHeader("Date", fmt::format("{:%a, %d %b %Y %H:%M:%S %Z}", tm));
+        if (req->RequestUrl() == "/plaintext") {
+          response->MutableBody() = "Hello, World!";
+        } else if (req->RequestUrl() == "/json") {
+          response->InsertHeader("Content-Type", "application/json");
+          response->MutableBody() = "{\"message\":\"Hello, World!\"}";
```

Comment:
For the JSON test, you must instantiate an object on each request.

See examples:

https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/C%2B%2B/cppcms/src/test_simple.cpp#L17

https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/C%2B%2B/drogon/drogon_benchmark/controllers/JsonCtrl.cc#L7

Reply:
done. thanks again.